pages/glossary/visible-inner-text.md

```diff
@@ -0,0 +1,38 @@
+# Visible inner text
+
+(The "visible inner text" defined here is similar to, but not the same as, [visible text content][].)
+
+The <dfn>visible inner text of a node</dfn> depends on the kind of node. 
+
+The <dfn id="for-text">visible inner text of a [text node][]</dfn> is:
+-   if the [text node][] is [visible][], its visible inner text is its [data][];
+-   if the [text node][] is not-[visible][], [rendered][], and contains only [whitespace][], its visible inner text is the string `" "` (a single ASCII whitespace);
+-   otherwise, the visible inner text of the [text node][] is the empty string (`""`).
+
+
+The <dfn id="for-element">visible inner text of an [element][]</dfn> is:
+-   if the [element][] is not [rendered][], its visible inner text is the empty string (`""`);
+-   if the [element][] is a [`<br>`][<br>] element, its visible inner text is a newline character (`"\n"`);
+-   if the [element][] is a [`<p>`][<p>] element, its visible inner text is the concatenation a newline character (`"\n"`) plus the visible inner text of its children (in [tree order][] in the [flat tree][]), plus a newline character (`"\n"`);
```

Comment:
Should that still be the case if the `p` element has `display: inline`?

Reply:
Good catch.  I see two ways to fix it:
1) in the "`<p>`" clause here, add an exception for "display: inline", or
2) remove this "`<p>`" clause so that a `<p>` without "display: inline" will be covered by the "block" clause further down, and a `<p>` with "display: inline" will be covered by the "otherwise" clause further down.   Right?

Which is better? 

Interesting tangent: for HTML like this: `<div><p style="display: inline">x</p><p style="display: inline">123</p><p style="display: inline">y</p></div>` browser/JS computes "innerText" of the `<div>` to be 'x\n\n123\n\ny'.   (I made a [test page for this](https://dan-tripp-siteimprove.github.io/act-rules-p-element-display-inline-innertext.html) here.)    I don't know what to make of this.

Reply:
🤔 It seems that [HTML innerText](https://html.spec.whatwg.org/multipage/dom.html#the-innertext-idl-attribute) does have a specific case for `p` elements (Step 8 of the second algorithm).
I'm not sure what to make of this either…
```html
<div role="button" aria-label="x123y">
  <p style="display: inline">x</p><p style="display: inline">123</p><p style="display: inline">y</p>
</div>
```
Do we want that to pass the rule or fail it? Looking at your test page, it seems the visible label is indeed `x123y` so this should pass the rule.

Reply:
I think we want it to pass the rule.   So I addressed it in https://github.com/act-rules/act-rules.github.io/pull/2075/commits/cde4ad480cfbceb1c43eb76ce22536f28b8f4480 through my option #2 above ("remove this "&lt;p&gt;" clause...") and a new passed example.